### PR TITLE
✨ Added function to print SiDB layout to svg 

### PR DESCRIPTION
**Pull request by Drewniok**
## Description

This PR adds a function which allows to print an SiDB cell-level layout and a charge distribution surface to a svg in the style of _SiQAD_.

CDS in dark mode and cell-level layout in light mode:

test_layout_dark_cds

test_layout_bright

## Checklist:

- [x] The pull request only contains commits that are related to it.
- [x] I have added appropriate tests and documentation.
- [ ] I have added a changelog entry.
- [ ] I have created/adjusted the Python bindings for any new or updated functionality.
- [x] I have made sure that all CI jobs on GitHub pass.
- [ ] The pull request introduces no new warnings and follows the project's style guidelines.

### CODE REVIEW

File: bindings/pyfiction/include/pyfiction/inout/write_qca_layout_svg.hpp [hunk 1]
```diff
@@ -26,9 +26,7 @@ inline void write_svg_layout(pybind11::module& m)
                                                      DOC(fiction_write_qca_layout_svg_params))
         .def(py::init<>())
         .def_readwrite("simple", &fiction::write_qca_layout_svg_params::simple,
-                       DOC(fiction_write_qca_layout_svg_params_simple))
-
-        ;
+                       DOC(fiction_write_qca_layout_svg_params_simple));
```
Comment (marcelwa) [inline bindings/pyfiction/include/pyfiction/inout/write_qca_layout_svg.hpp:29]: Please revert this change

File: bindings/pyfiction/test/inout/test_write_sidb_layout_to_svg_in_siqad_layout.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    svg (str): The input SVG string to be normalized.
+
+    Returns:
+    str: A string with all whitespace characters removed.
+    """
+    # Remove all whitespace (spaces, tabs, newlines)
+    return ''.join(svg.split())
+
+cell_level_dark_mode = """<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!-- Generated by fiction v0.6.5 (https://github.com/cda-tum/fiction) -->
```
Comment (marcelwa) [inline bindings/pyfiction/test/inout/test_write_sidb_layout_to_svg_in_siqad_layout.py:22]: These strings hardcode the fiction version and repo. These should be set dynamically or these tests will fail on the next update. See the existing C++ tests for file writing. Maybe we have to expose global variables like  `__version__` and `__repo__` in Python first, though. It's easy enough since C++ already sets them dynamically. We can simply bind those to Python variables.

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp
Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:1]: The respective QCA writer is called `write_qca_layout_svg`. I feel like these should be consistent with each other. Would you mind renaming your file, functions, include guards, and bindings?

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+
+// SVG path definitions for lattice points and SiDBs
+static constexpr const char* PATH_DEFINITION_TEMPLATE = R"(
+<defs>
+    <!-- lattice point -->
+    <circle id="lattice_point" cx="0" cy="0" r="{0}"/>
+    <!-- SiDB -->
+    <circle id="sidb_color" cx="0" cy="0" r="{1}"/>
+</defs>
+)";
```
Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:77]: Can be `inline constexpr` instead

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+     * Dark background.
+     */
+    DARK,
+    /**
+     * Light background.
+     */
+    LIGHT
+};
+/**
+ * Parameters for writing SiDB layouts in SVG format.
```
Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:96]:
```suggestion
 * Parameters for writing SiDB layouts to SVG format.
```

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp [hunk 5]
```diff
@@ -0,0 +1,417 @@
...
+     */
+    double lattice_point_size = 0.3;
+    /**
+     * Size of the SiDB in SVG units.
+     */
+    double sidb_size = 0.9;
+    /**
+     * Edge width of the SiDB.
+     */
+    double sidb_edge_width = 0.3;
```
Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:111]: What is an edge width?

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp [hunk 6]
```diff
@@ -0,0 +1,417 @@
...
+enum class color_mode : uint8_t
+{
+    /**
+     * Dark background.
+     */
+    DARK,
+    /**
+     * Light background.
+     */
+    LIGHT
```
Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:93]: Does it only affect the background?

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp [hunk 7]
```diff
@@ -0,0 +1,417 @@
...
+            background_color = fiction::svg::BACKGROUND_COLOR_DARK;
+            sidb_color       = fiction::svg::SIDB_DOT_DARK_MODE;
+            sidb_edge_color  = fiction::svg::SIDB_DOT_LINE_COLOR_DARK_MODE;
+        }
+    }
+
+    /**
+     * Generates an SVG string representing an H-Si lattice point.
+     */
+    [[nodiscard]] std::string generate_lattice_point(double x, double y, const std::string& fill_color) const noexcept
```
Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:194]:
```suggestion
    [[nodiscard]] std::string generate_lattice_point(const double x, const double y, const std::string& fill_color) const noexcept
```

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp [hunk 8]
```diff
@@ -0,0 +1,417 @@
...
+            // fiction namespace is needed to avoid windows and linux conflict
+            background_color = fiction::svg::BACKGROUND_COLOR_DARK;
+            sidb_color       = fiction::svg::SIDB_DOT_DARK_MODE;
+            sidb_edge_color  = fiction::svg::SIDB_DOT_LINE_COLOR_DARK_MODE;
+        }
+    }
+
+    /**
+     * Generates an SVG string representing an H-Si lattice point.
+     */
```
Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:193]: Missing parameters and return in docstring

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp [hunk 9]
```diff
@@ -0,0 +1,417 @@
...
+    [[nodiscard]] std::string generate_lattice_point(double x, double y, const std::string& fill_color) const noexcept
+    {
+        return fmt::format(R"(<use xlink:href="#lattice_point" x="{0}" y="{1}" style="fill:{2};"/>)", x, y, fill_color);
+    }
+
+    /**
+     * Generates an SVG string representing an SiDB.
+     */
```
Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:201]: Missing parameters and return in docstring

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp [hunk 10]
```diff
@@ -0,0 +1,417 @@
...
+            R"(<use xlink:href="#sidb_color" x="{0}" y="{1}" style="fill:{2}; fill-opacity:{3}; stroke:{4}; stroke-width:{5};"/>)",
+            x, y, fill_color, fill_opacity, border_color, ps.sidb_edge_width);
+    }
+
+    /**
+     * Generates the SVG layout with both H-Si lattice points and SiDBs.
+     */
+    void generate_svg() const
+    {
+        std::stringstream svg_content;
```
Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:250]:
```suggestion
        std::stringstream svg_content{};
```

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp [hunk 11]
```diff
@@ -0,0 +1,417 @@
...
+
+        // Prepare the formatted PATH_DEFINITION with the sizes
```
Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:252]:
```suggestion
        // Prepare the formatted PATH_DEFINITION_TEMPLATE with the sizes
```

Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:252]: What do you mean by "formatted"? Don't you only format it here?

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp [hunk 12]
```diff
@@ -0,0 +1,417 @@
...
+/**
+ * Writes an SVG representation of a cell-level SiDB layout into an output stream.
+ *
+ * @tparam Lyt SiDB cell-level layout type.
+ * @param lyt The layout to be written.
+ * @param os The output stream to write into.
+ * @param ps Parameters.
+ *
+ * @note SiDB defects are not supported yet.
+ */
```
Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:345]: Please move notes to directly after the explanation
```suggestion
/**
 * Writes an SVG representation of a cell-level SiDB layout into an output stream.
 *
 * @note SiDB defects are not supported yet.
 *
 * @tparam Lyt SiDB cell-level layout type.
 * @param lyt The layout to be written.
 * @param os The output stream to write into.
 * @param ps Parameters.
 */
```

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp [hunk 13]
```diff
@@ -0,0 +1,417 @@
...
+
+/**
+ * Writes an SVG representation of a cell-level SiDB layout into a file.
+ *
+ * @tparam Lyt SiDB cell-level layout type.
+ * @param lyt The layout to be written.
+ * @param filename The file name to create and write into.
+ * @param ps Parameters.
+ *
+ * @note SiDB defects are not supported yet.
```
Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:366]: Here as well

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp [hunk 14]
```diff
@@ -0,0 +1,417 @@
...
+        // Generate the final SVG content
+        os << fmt::format(fiction::svg::HEADER_TEMPLATE, FICTION_VERSION, FICTION_REPO, viewbox_x, viewbox_y,
+                          viewbox_width, viewbox_height, formatted_path_definition, background_rect, svg_content.str());
+    }
+};
+
+}  // namespace detail
+
+/**
+ * Writes an SVG representation of a cell-level SiDB layout into an output stream.
```
Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:337]: We usually say "SiDB cell-level layout"

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp [hunk 15]
```diff
@@ -0,0 +1,417 @@
...
+    static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
+    static_assert(has_sidb_technology_v<Lyt>, "Lyt must be a SiDB layout");
+    static_assert(!is_sidb_defect_surface_v<Lyt>, "SiDB defects are not supported");
+
+    detail::write_sidb_layout_svg_impl<Lyt> p{lyt, os, ps};
+    p.run();
+}
+
+/**
+ * Writes an SVG representation of a cell-level SiDB layout into a file.
```
Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:359]: Here too

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp [hunk 16]
```diff
@@ -0,0 +1,417 @@
...
+ * result as a string. The SVG is generated according to SiQAD style parameters and is
+ * suitable for visualization.
+ *
+ * @tparam Lyt SiDB cell-level layout type.
+ * @param lyt The input layout to be converted to an SVG string.
+ * @param ps Optional parameters to customize the SVG generation process.
+ *           Defaults to an instance of `write_sidb_layout_svg_in_siqad_style_params`.
+ * @return A string containing the SVG representation of the provided SiDB layout.
+ *
+ */
```
Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:397]: Why does this docstring differ from the others in style and content? Also, the note is missing.

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp [hunk 17]
```diff
@@ -0,0 +1,417 @@
...
+template <typename Lyt>
+std::string write_sidb_layout_to_svg_and_return_as_string(const Lyt&                                         lyt,
+                                                          const write_sidb_layout_svg_in_siqad_style_params& ps = {})
```
Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:400]: Why is this function required in the first place? We have the output stream function already. That can be used to generate a string via a `std::stringstream`.

File: include/fiction/io/write_svg_layout.hpp [hunk 18]
```diff
@@ -511,7 +512,7 @@ class write_qca_layout_svg_impl
                     {
                         cell_descriptions << fmt::format(
                             desc_col.first, desc_col.second,
-                            svg::STARTING_OFFSET_TILE_X + svg::STARTING_OFFSET_CELL_X + c.x * svg::CELL_DISTANCE,
+                            svg::STARTING_OFFSET_TILE_X + svg::STARTING_OFFSET_CELL_X + (c.x * svg::CELL_DISTANCE),
```
Comment (marcelwa) [inline include/fiction/io/write_svg_layout.hpp:515]: Why did you add parentheses here but not around `c.y * svg::CELL_DISTANCE`?

File: test/io/write_sidb_layout_to_svg_in_siqad_style.cpp [hunk 19]
```diff
@@ -0,0 +1,466 @@
...
+
+/**
+ * This function takes an SVG string as input and returns a new string with all
+ * whitespace characters removed. The purpose is to create a compact representation
+ * of the SVG content, which can be useful for comparisons, optimizations, or transmission.
+ *
+ * @param svg The input SVG string to be normalized.
+ * @return A string with all whitespace characters removed.
+ */
+auto normalize_svg = [](const std::string& svg) noexcept
```
Comment (marcelwa) [inline test/io/write_sidb_layout_to_svg_in_siqad_style.cpp:33]: Objects in global namespace should at least be `const`, even better also `static`, even better yet `constexpr`. But also, why is this a lambda instead of a function?

File: test/io/write_sidb_layout_to_svg_in_siqad_style.cpp [hunk 20]
```diff
@@ -0,0 +1,466 @@
...
+{
+    std::string result = svg;
+
+    // Remove all whitespace (spaces, tabs, newlines)
+    result.erase(std::remove_if(result.begin(), result.end(), ::isspace), result.end());
+
+    return result;
+};
+
+const std::string EXPECTED_SVG_LIGHT_CELL_LEVEL = fmt::format(R"(<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (marcelwa) [inline test/io/write_sidb_layout_to_svg_in_siqad_style.cpp:43]: Same rules as above apply to all these global constants.

File: bindings/pyfiction/include/pyfiction/inout/write_sidb_layout_to_svg_in_siqad_style.hpp
Comment (marcelwa) [inline bindings/pyfiction/include/pyfiction/inout/write_sidb_layout_to_svg_in_siqad_style.hpp:1]: Please add the `DOC(...)` docstrings

**Review by marcelwa — CHANGES_REQUESTED** [review]

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp (hunk 6 above)
Comment (Drewniok) [reply include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp:93]: You are rihgt. It is more than that.

File: test/io/write_sidb_layout_to_svg_in_siqad_style.cpp (hunk 20 above)
Comment (Drewniok) [reply test/io/write_sidb_layout_to_svg_in_siqad_style.cpp:43]: @marcelwa, ``constexpr`` does not work here, right?

File: include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp (hunk 17 above)
Comment (Drewniok) [reply include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp]: Sorry, you are right! I had some problems with the Python bindings, but I have it fixed now.

File: test/io/write_sidb_layout_to_svg_in_siqad_style.cpp (hunk 20 above)
Comment (marcelwa) [reply test/io/write_sidb_layout_to_svg_in_siqad_style.cpp]: Unfortunately, `std::string` cannot be `constexpr` indeed. If `fmt::format` supports it, however, you could try changing the type of this expression to `auto` (or `constexpr const char*`) and append a `.c_str()` to the `fmt::format` call. Not a guarantee that it works, but worth a shot :man_shrugging:

File: bindings/pyfiction/include/pyfiction/utils/version_info.hpp [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+using namespace pybind11::literals;
+
+inline void version_info(pybind11::module& m)
+{
+    m.attr("FICTION_VERSION") = fiction::FICTION_VERSION;
+    m.attr("FICTION_REPO")    = fiction::FICTION_REPO;
+    m.attr("COMPILED_DATE")   = fiction::COMPILED_DATE;
+    m.attr("COMPILED_TIME")   = fiction::COMPILED_TIME;
+
+    m.doc() = "Python bindings for the fiction project version information.";
```
Comment (marcelwa) [inline bindings/pyfiction/include/pyfiction/utils/version_info.hpp:24]: As far as I understand, this sets a global docstring for the module `mnt.pyfiction` and thus overrides
```c++
m.doc() = "Python bindings for fiction, a framework for Design Automation for Field-coupled Nanotechnologies";
```
from `pyfiction.cpp`.

File: bindings/pyfiction/include/pyfiction/utils/version_info.hpp [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+{
+
+using namespace pybind11::literals;
+
+inline void version_info(pybind11::module& m)
+{
+    m.attr("FICTION_VERSION") = fiction::FICTION_VERSION;
+    m.attr("FICTION_REPO")    = fiction::FICTION_REPO;
+    m.attr("COMPILED_DATE")   = fiction::COMPILED_DATE;
+    m.attr("COMPILED_TIME")   = fiction::COMPILED_TIME;
```
Comment (marcelwa) [inline bindings/pyfiction/include/pyfiction/utils/version_info.hpp:22]: I believe Python has a standardized form for these variables like `__version__`, `__url__`, etc. Could you look this up and set the names accordingly?

File: include/fiction/io/write_sidb_layout_svg.hpp
Comment (marcelwa) [inline include/fiction/io/write_sidb_layout_svg.hpp:1]: It would be awesome to add this functionality to the CLI as well when executing `show -c`. You can find the code to adjust in `cli/stores.hpp` lines 456 to 492.

**Review by marcelwa — CHANGES_REQUESTED** [review]

File: bindings/pyfiction/include/pyfiction/utils/version_info.hpp (hunk 22 above)
Comment (Drewniok) [reply bindings/pyfiction/include/pyfiction/utils/version_info.hpp:22]: oh, you are right! That's a good point.

File: cli/stores.hpp [hunk 23]
```diff
@@ -462,9 +463,9 @@ void show<fiction::cell_layout_t>(std::ostream& os, const fiction::cell_layout_t
     {
         using Lyt = typename std::decay_t<decltype(lyt_ptr)>::element_type;
 
-        if constexpr (!fiction::has_qca_technology_v<Lyt>)
+        if constexpr (!fiction::has_qca_technology_v<Lyt> && !fiction::has_sidb_technology_v<Lyt>)
         {
-            cmd.env->out() << fmt::format("[e] {} is not a QCA layout", lyt_ptr->get_layout_name()) << std::endl;
+            cmd.env->out() << fmt::format("[e] {} is either a QCA layout nor an SiDB layout", lyt_ptr->get_layout_name()) << std::endl;
```
Comment (marcelwa) [inline cli/stores.hpp]:
```suggestion
            cmd.env->out() << fmt::format("[e] {} is neither a QCA layout nor an SiDB layout", lyt_ptr->get_layout_name()) << std::endl;
```

**Review by marcelwa — CHANGES_REQUESTED** [review]

File: include/fiction/io/write_sidb_layout_svg.hpp [hunk 24]
```diff
@@ -0,0 +1,400 @@
...
+template <typename Lyt>
+void write_sidb_layout_svg(const Lyt& lyt, std::ostream& os, const write_sidb_layout_svg_params& ps = {})
+{
+    static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
+    static_assert(has_sidb_technology_v<Lyt>, "Lyt must be a SiDB layout");
+    static_assert(!is_sidb_defect_surface_v<Lyt>, "SiDB defects are not supported");
+
+    detail::write_sidb_layout_svg_impl<Lyt> p{lyt, os, ps};
+    p.run();
+}
```
[the remark this answers is not in the document]
Comment (Drewniok) [reply include/fiction/io/write_sidb_layout_svg.hpp:367]: Yes, it is. I have added an appropriate test.

File: include/fiction/io/write_svg_layout.hpp
Comment (marcelwa) [inline include/fiction/io/write_svg_layout.hpp:1]: There seems to be a mismatch still with respect to file names and function names. We have `write_svg_layout.hpp` and `write_sidb_layout_svg.hpp` in C++, but `write_qca_layout_svg.hpp` and `write_sidb_layout_svg.hpp` in Python. The only option to streamline this without violate the current API is to put the new SVG function for SiDB layouts in `write_svg_layout.hpp`.

File: bindings/pyfiction/include/pyfiction/inout/write_sidb_layout_svg.hpp [hunk 25]
```diff
@@ -0,0 +1,85 @@
...
+        .def_readwrite("lattice_point_size", &fiction::write_sidb_layout_svg_params::lattice_point_size,
+                       DOC(fiction_write_sidb_layout_svg_params_lattice_point_size))
+        .def_readwrite("sidb_size", &fiction::write_sidb_layout_svg_params::sidb_size,
+                       DOC(fiction_write_sidb_layout_svg_params_sidb_size))
+        .def_readwrite("sidb_border_width", &fiction::write_sidb_layout_svg_params::sidb_border_width,
+                       DOC(fiction_write_sidb_layout_svg_params_sidb_border_width))
+        .def_readwrite("color_background", &fiction::write_sidb_layout_svg_params::color_background,
+                       DOC(fiction_write_sidb_layout_svg_params_color_background));
+
+    detail::write_sidb_layout_svg_impl<py_charge_distribution_surface_100>(m);
```
[the remark this answers is not in the document]
Comment (Drewniok) [reply bindings/pyfiction/include/pyfiction/inout/write_sidb_layout_svg.hpp:74]: Good catch! Thank you!

File: cli/stores.hpp [hunk 26]
```diff
@@ -462,9 +463,11 @@ void show<fiction::cell_layout_t>(std::ostream& os, const fiction::cell_layout_t
...
 
-        if constexpr (!fiction::has_qca_technology_v<Lyt>)
+        if constexpr (!fiction::has_qca_technology_v<Lyt> && !fiction::has_sidb_technology_v<Lyt>)
         {
-            cmd.env->out() << fmt::format("[e] {} is not a QCA layout", lyt_ptr->get_layout_name()) << std::endl;
+            cmd.env->out() << fmt::format("[e] {} is neither a QCA layout nor an SiDB layout",
+                                          lyt_ptr->get_layout_name())
+                           << std::endl;
         }
         else if constexpr (!std::is_same_v<fiction::coordinate<Lyt>, fiction::offset::ucoord_t>)
```
[the remark this answers is not in the document]
Comment (Drewniok) [reply cli/stores.hpp:472]: Oh, you are right!  I was too quick on that one!

File: bindings/pyfiction/include/pyfiction/inout/write_svg_layout.hpp [hunk 27]
```diff
@@ -17,25 +17,79 @@
...
-        ;
+                       DOC(fiction_write_qca_layout_svg_params_simple));
+    ;
 
-    void (*write_svg_layout_function_pointer)(const py_qca_layout&, const std::string_view&,
-                                              const fiction::write_qca_layout_svg_params&) =
-        &fiction::write_qca_layout_svg<py_qca_layout>;
+    detail::write_svg_layout_impl<py_charge_distribution_surface_100>(m);
+    detail::write_svg_layout_impl<py_charge_distribution_surface_100>(m);
+    detail::write_svg_layout_impl<py_charge_distribution_surface>(m);
```
Comment (marcelwa) [inline bindings/pyfiction/include/pyfiction/inout/write_svg_layout.hpp:88]:
```suggestion
    detail::write_svg_layout_impl<py_charge_distribution_surface_111>(m);
    detail::write_svg_layout_impl<py_charge_distribution_surface_100>(m);
    detail::write_svg_layout_impl<py_charge_distribution_surface>(m);
```

Still the same copy-paste-error.

File: test/io/write_sidb_layout_svg.cpp
Comment (marcelwa) [inline test/io/write_sidb_layout_svg.cpp:1]: I don't think we have a dedicated test file for the QCA SVG writer, unfortunately. However, I would like to stay consistent with the file naming scheme anyway. Could you thus rename this file to `write_svg_layout.cpp`?

File: include/fiction/io/write_svg_layout.hpp [hunk 28]
```diff
@@ -37,6 +43,58 @@ struct write_qca_layout_svg_params
...
+enum class color_mode : uint8_t
+{
+    /**
+     * Dark mode.
+     */
+    DARK,
+    /**
+     * Light mode.
+     */
+    LIGHT
```
Comment (marcelwa) [inline include/fiction/io/write_svg_layout.hpp:58]: Could you indicate here what exactly changes in the two modes? Background color, dot color, etc.

Comment (marcelwa) [reply include/fiction/io/write_svg_layout.hpp:58]: Since we don't have it for QCA, you might as well move it into `write_sidb_layout_svg_params`.

File: include/fiction/io/write_svg_layout.hpp [hunk 29]
```diff
@@ -37,6 +43,58 @@ struct write_qca_layout_svg_params
...
+{
+    /**
+     * Parameters for writing SVG QCA layouts.
+     */
+    write_qca_layout_svg_params qca_params;
+    /**
+     * Parameters for writing SiDB layouts to SVG format.
+     */
+    write_sidb_layout_svg_params sidb_params;
+};
```
Comment (marcelwa) [inline include/fiction/io/write_svg_layout.hpp:96]: Why do we need this? Does it make sense to group these two completely different parameter sets in one object?

Comment (marcelwa) [reply include/fiction/io/write_svg_layout.hpp:96]: I would argue, each SVG function should have its own set of parameters.

File: include/fiction/io/write_svg_layout.hpp [hunk 30]
```diff
@@ -771,6 +1139,116 @@ void write_qca_layout_svg(const Lyt& lyt, const std::string_view& filename, cons
...
+    else if constexpr (has_sidb_technology_v<Lyt>)
+    {
+        write_sidb_layout_svg(lyt, os, ps.sidb_params);
+    }
+    else
+    {
+        throw std::invalid_argument("Unknown layout type");
+    }
+    os.close();
+}
```
Comment (marcelwa) [inline include/fiction/io/write_svg_layout.hpp:1250]: What do we need these generic functions for?

File: bindings/pyfiction/include/pyfiction/inout/write_svg_layout.hpp [hunk 31]
```diff
@@ -32,25 +29,35 @@ inline void write_sidb_layout_svg_impl(pybind11::module& m)
...
         "layout"_a, "ps"_a = fiction::write_sidb_layout_svg_params{}, DOC(fiction_write_sidb_layout_svg));
+
+
+    // QCA plot
+    void (*write_qca_layout_svg_pointer)(const py_qca_layout&, const std::string_view&,
+                                              const fiction::write_qca_layout_svg_params&) =
+        &fiction::write_qca_layout_svg<py_qca_layout>;
+
+    m.def("write_qca_layout_svg", write_qca_layout_svg_pointer, "layout"_a, "filename"_a,
+          "params"_a = fiction::write_qca_layout_svg_params{}, DOC(fiction_write_qca_layout_svg));
```
Comment (marcelwa) [inline bindings/pyfiction/include/pyfiction/inout/write_svg_layout.hpp]: Since the QCA SVG writer is defined in this function, it is called and thus registered six times. That could lead to problems. Rather make two `_impl` functions. One for QCA and one for SiDB.

**Review by marcelwa — CHANGES_REQUESTED** [review]

File: include/fiction/io/write_svg_layout.hpp (hunk 30 above)
Comment (Drewniok) [reply include/fiction/io/write_svg_layout.hpp:1250]: We do not need it. I thought it would be nice since we do it similiar in the ``print_layout`` function.

**Review by marcelwa — APPROVED** [review]
